_adr/80/index.adoc

```diff
@@ -1,22 +1,89 @@
+organisation.  In order to assist the customers, members of {glossary-customersupport} require a mechanism that allows them to capture the state of the
+instance of the service for diagnostic purposes.
+
+{glossary-customersupport} will use the gathered information in conjunction with the problem statement provided by the customer in order to efficiently
+investigate the problem and determine a solution.
 
-The data gathering mechanism must consider the issue of Personally Identifiable Information (PII) or other sensitive
-information and ensure that it is handled appropriately.
+The data gathering mechanism must consider the issue of Personally Identifiable Information (PII) or other sensitive information and ensure
+that it is handled appropriately.
 
 Authorization controls must be in place to ensure that the diagnostic gathering mechanism can only be used by suitably
 permissioned groups.
 
+## Goals
+
+* Identify the preferred solution for the diagnostic gathering mechanism.
+* Provide guidelines for content that should be included.
+
+## Non-goals
+
+None
+
+## Stakeholders
+* {glossary-businessstakeholder}
+* {glossary-sre}
+* {glossary-customersupport}
+
+## Current Architecture
+
+None
+
+## Proposed Architecture
+
+https://datatracker.ietf.org/doc/html/rfc2119[RFC-2119] keywords are used in this proposal.
+
+It is RECOMMENDED that managed services use https://source.redhat.com/groups/public/openshiftplatformsre/wiki/backplane_scripts[Back Plane]
+https://github.com/openshift/managed-scripts/[managed scripts] as the foundation of the data gathering mechanism. Backplane has an
+established https://source.redhat.com/groups/public/openshiftplatformsre/wiki/backplane_cluster_permissions[authorization model] that ensures
+that members of the {glossary-customersupport} have the permissions to run the scripts.
+
+Each Managed Service that implements this ADR MUST provide one or more backplane scripts that interrogate the state of a service instance.
+The exact way way the script will interrogate state will be service specific, however if the service provided on top of OpenShift, it is
+RECOMMENDED that the script invoke the `oc adm inspect` command to gather Kubernetes state in addition to any other service specific
+state gathering commands.
+
+It is RECOMMENDED that the output from the backplane script be a UNIX tar stream.  This will allow the artifact to provide the diagnostic
+information in a structured form that can easily be processed using common command-line tools.
+
+It is RECOMMENDED that the script takes reasonable steps to avoid the collection of PII or other sensitive information. This may be done
+either by redacting artifacts known to contain secrets/keys etc, or not gathering certain information in the first place.  For
+example, in a Java based system where PII may be present in the Java heap, the safe option is not to collect heap dumps.
```

Comment:
While I understand the motivation for this, do we really understand and accept the implications? I know we don't have a good mechanism for collecting heap dumps right now, but this basically means no heap dumps, even when OOME has occurred. That rules out the main problem finding technique for a whole class of problem. This rule also prevents us from taking log dumps, for example, which again could severely limit our ability to fix problems, and understand root causes. 

I would much rather we had the contractual wording to collect this kind of diagnostic data if we really had to (even if it's not collected by default), even if we're obliged to delete the information within a certain timeframe of taking it. For example, what if there was a mechanism to identify files in the tar.gz which contained PII and those files were then managed in a way that was GDPR compliant? I assume you've investigated that route and found it's not possible, or we just can't go there yet, so I don't see this as a blocker, just a source of continuing headaches.

Reply:
I think I understand your points. How about I reword along the lines of?

```suggestion
By default, it is RECOMMENDED that the scripts takes reasonable steps to avoid the collection of PII or other sensitive information. This may be done either by redacting artifacts known to contain secrets/keys etc, or not gathering certain information in the first place. The scripts MAY offer alternative modes of operations that collect deeper information.  The RECOMMENDED mechanism to enable these modes is to make use of [managed script parameters](https://github.com/openshift/managed-scripts/blob/main/scripts/SREP/example/metadata.yaml#L30).  If the use of such a mode may result in a artifacts containing PII or other sensitive information, the parameter's description MUST document that fact.
```

In the case where the user elects to use the special modes, the responsibility would be on them to treat the artefact in a manner compliant with the Organisation's PII data processing standards.

Reply:
@tombentley would the proposed reword address your comment?

Reply:
Speaking as an engineer, "yes" that would scratch my itch of being able to get the information I might need in order to understand problems manifesting in production. 

But thinking about the PoV of some GDPR/InfoSec persona, "no" because I would want backplane to provide mechanisms for knowing when SREs use these options and ensuring that the information gathered through them were managed in accordance with national laws and organisational policies.

In other words, I don't think there can be easy technical solutions to this: We need to have a conversation with business, legal, SRE and dev representation to figure out what might be possible if backplane were able to have first-class support for collecting and managing sensitive information, or the acceptance that there will be circumstances in which we won't be able to fully investigate production problems (and be prepared to tell the end customer that "there's a problem, we don't understand it and our company policy on being able to collect possibly-sensitive data about the service we're running for you prevents us from getting to the root cause").

There _might_ be hard technical solutions to this. For example if customers' usage of the cluster is such that the topics contain only encrypted data (and the investigating engineers cannot access the keys) then there can be no leakage of sensitive data from doing a heap dump.